### ISSUE TITLE
Building binaryen fails

### ISSUE THREAD

**Issue by Moeplhausen** [kripken/emscripten#5637]
I have switched from emscripten 1.37.18 to 1.37.21 (both archlinux packages).

After that the command `embuilder.py build binaryen` stopped working. Error:

```
[ 34%] Building CXX object src/passes/CMakeFiles/passes.dir/DeadCodeElimination.cpp.o
/home/m/.emscripten_ports/binaryen/binaryen-version_37/src/wasm/wasm-s-parser.cpp: In member function 'wasm::Expression* wasm::SExpressionWasmBuilder::makeExpression(wasm::Element&)':
/home/m/.emscripten_ports/binaryen/binaryen-version_37/src/wasm/wasm-s-parser.cpp:871:9: error: this statement may fall through [-Werror=implicit-fallthrough=]
         if (!strncmp(str, "wake", strlen("wake"))) return makeAtomicWake(s);
         ^~
/home/m/.emscripten_ports/binaryen/binaryen-version_37/src/wasm/wasm-s-parser.cpp:873:7: note: here
       default: abort_on(str);
       ^~~~~~~
[ 35%] Building CXX object src/passes/CMakeFiles/passes.dir/DuplicateFunctionElimination.cpp.o
[ 36%] Building CXX object src/passes/CMakeFiles/passes.dir/ExtractFunction.cpp.o
[ 38%] Building CXX object src/passes/CMakeFiles/passes.dir/FlattenControlFlow.cpp.o
[ 39%] Building CXX object src/passes/CMakeFiles/passes.dir/Inlining.cpp.o
cc1plus: all warnings being treated as errors
```

I used both `emcc --clear-ports `and `emcc --clear-cache` after switching to the new emscripten version. It just won't compile. Emscripten version 1.37.18 still works though.

**Comment by kripken** [comment]
This should be fixed in 1.37.22 which was just tagged, please verify.

**Comment by Moeplhausen** [comment]
Thanks for the quick fix. It works now.
